terraform/environments/delius-core/modules/components/oracle_db_shared/iam.tf

```diff
@@ -180,4 +180,59 @@ resource "aws_iam_policy" "db_access_to_secrets_manager" {
 resource "aws_iam_role_policy_attachment" "db_access_to_secrets_manager" {
   role       = aws_iam_role.db_ec2_instance_iam_role.name
   policy_arn = aws_iam_policy.db_access_to_secrets_manager.arn
+}
+
+# new IAM role OEM setup to allow ec2s to access secrets manager and kms keys
+resource "aws_iam_role" "db_ec2_oem_role" {
+  name                  = "EC2OracleEnterpriseManagementSecretsRole"
+  assume_role_policy    = data.aws_iam_policy_document.ec2_oem_assume_policy.json
+}
+
+data "aws_iam_policy_document" "ec2_oem_assume_policy" {
+  statement {
+    effect  = "Allow"
+    actions = [
+      "sts:AssumeRole"
+    ]
+    principals {
+      type        = "AWS"
+      identifiers = ["*"]
+    }
+  }
+}
+
+resource "aws_iam_role_policy_attachment" "allow_new_role_secrets_manager" {
+  role       = aws_iam_role.db_ec2_oem_role.name
+  policy_arn = aws_iam_policy.db_access_to_secrets_manager.arn
+}
+
+resource "aws_iam_role_policy_attachment" "allow_kms_keys_access" {
+  role       = aws_iam_role.db_ec2_oem_role.name
+  policy_arn = aws_iam_policy.business_unit_kms_key_access.arn
+}
+
+# Define a custom IAM policy document to grant assume role permissions
+data "aws_iam_policy_document" "ec2_oem_assume_role_policy_document" {
```

Comment:
I'm not sure you need this or the following 2 resources ev2_oem_assume_role_policy or assume_role_policy_attachment.  At least, we haven't added an explicit Assume Policy that I can see to the instance role and it seems to work OK.

Reply:
This policy is attached to the existing role in [here](https://github.com/ministryofjustice/modernisation-platform-environments/blob/2c6b4a7919224cb4de5b517a7e5efd6673c7dc6b/terraform/environments/delius-core/modules/components/oracle_db_shared/iam.tf#L235).

Reply:
I have removed this resource now as not needed anymore.